core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -96,7 +96,16 @@ void SYCLInternal::initialize(const cl::sycl::device& d) {
   const bool ok_init = nullptr == m_scratchSpace || nullptr == m_scratchFlags;
   const bool ok_dev  = true;
   if (ok_init && ok_dev) {
-    m_queue = std::make_unique<cl::sycl::queue>(d);
+    auto exception_handler = [](sycl::exception_list exceptions) {
+      for (std::exception_ptr const& e : exceptions) {
+        try {
+          std::rethrow_exception(e);
+        } catch (sycl::exception const& e) {
+          Kokkos::abort(e.what());
+        }
```

Comment:
@dhollman says:
> I don't understand this completely. Are all of the exceptions in a sycl::exception_list derived from sycl::exception, or does the behavior of the handler depend on the order of the exceptions and throw out of itself sometimes?

Reply:
Yes, they all derive from `sycl::exception`. With this pull request, we just abort with a message for the first error that we see. 
I could be convinced to implement a different behavior if someone feels strongly about it.

Reply:
About the only reasonable thing to do would be to report all the exceptions before aborting.  I have a preference towards this as I don't like throwing away useful information.